### PR TITLE
Add a migration which detects and filters out unreferenced slabs

### LINKED ISSUE #5634: Migration to clean up unreferenced slabs
## Problem Definition

Some accounts in the execution state contain unreferenced slabs. They cause the atree storage health check to fail, which makes it difficult to assess if a migration was successful, as a health check to confirm the success after the migration will also fail.

## Proposed Solution

Implement a new storage migration, that filters all unreferenced slabs/payloads and writes them to a separate backup payloads file for potential future investigation.

We can then run it both before the atree register inlining migration, and before the Cadence 1.0 migration.

## Definition of Done

When integrated into migrations like the atree register inlining migration and the Cadence 1.0 migration, health checks before those migrations should not report any unreferenced slabs.

### PR DESCRIPTION
**Pull request by turbolent**
Closes #5634

### CODE REVIEW

File: cmd/util/ledger/migrations/filter_unreferenced_slabs_migration.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+				if _, ok := unreferencedRootSlabIDs[storageID]; ok {
+					filteredPayloads = append(filteredPayloads, payload)
+					continue
+				}
+			}
+
+			newPayloads = append(newPayloads, payload)
+		}
+
+		// TODO: write filtered payloads to a separate file
```
Comment (turbolent) [inline cmd/util/ledger/migrations/filter_unreferenced_slabs_migration.go:150]: @fxamacker @janezpodhostnik What would be a good way to write these filtered payloads? Should we create one payloads file per account? One file with all filtered payloads of all account?

Comment (janezpodhostnik) [reply cmd/util/ledger/migrations/filter_unreferenced_slabs_migration.go:150]: The report writer can be used.

I would create one file for all accounts with the report writer. The account address can be deduced from the payload key.

Comment (turbolent) [reply cmd/util/ledger/migrations/filter_unreferenced_slabs_migration.go:150]: @janezpodhostnik As far as I understand, the report writer generates JSON? Do you mean that we should  serialize the payloads to JSON (hex-encoded)?

**Comment by turbolent** [comment]
@fxamacker We actually need this on master right?

**Comment by turbolent** [comment]
Replaced by #5653

**Pull request closed without merging**